core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/externaldatabase/start/handler/StartExternalDatabaseHandler.java

```diff
@@ -0,0 +1,117 @@
+import com.sequenceiq.cloudbreak.reactor.api.event.externaldatabase.StartExternalDatabaseResult;
+import com.sequenceiq.cloudbreak.service.environment.EnvironmentClientService;
+import com.sequenceiq.environment.api.v1.environment.model.response.DetailedEnvironmentResponse;
+import com.sequenceiq.flow.reactor.api.handler.ExceptionCatcherEventHandler;
+
+import reactor.bus.Event;
+
+@Component
+public class StartExternalDatabaseHandler extends ExceptionCatcherEventHandler<StartExternalDatabaseRequest> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StartExternalDatabaseHandler.class);
+
+    @Inject
+    private ExternalDatabaseService startService;
+
+    @Inject
+    private StackUpdaterService stackUpdaterService;
+
+    @Inject
+    private EnvironmentClientService environmentClientService;
+
+    @Inject
+    private ExternalDatabaseConfig externalDatabaseConfig;
+
+    @Override
+    public String selector() {
+        return "StartExternalDatabaseRequest";
+    }
+
+    @Override
+    protected Selectable defaultFailureEvent(Long resourceId, Exception e, Event<StartExternalDatabaseRequest> event) {
+        Stack stack = event.getData().getStack();
+        LOGGER.error(String.format("Exception during DB 'start' for stack/cluster: %s", stack.getName()), e);
+        return startFailedEvent(stack, e);
+    }
+
+    @Override
+    protected Selectable doAccept(HandlerEvent event) {
+        LOGGER.debug("In StartExternalDatabaseHandler.doAccept");
+        StartExternalDatabaseRequest request = event.getData();
+        Stack stack = request.getStack();
+        DatabaseAvailabilityType externalDatabase = ObjectUtils.defaultIfNull(stack.getExternalDatabaseCreationType(), DatabaseAvailabilityType.NONE);
+        LOGGER.debug("External database: {} for stack {}", externalDatabase.name(), stack.getName());
+        LOGGER.debug("Getting environment CRN for stack {}", stack.getName());
+        DetailedEnvironmentResponse environment = environmentClientService.getByCrn(stack.getEnvironmentCrn());
+        Selectable result;
+        try {
+            if (StackType.WORKLOAD != stack.getType()) {
+                LOGGER.debug("External database start in Cloudbreak service is required for WORKLOAD stacks only.");
+                result = new StartExternalDatabaseResult(stack.getId(), EXTERNAL_DATABASE_STARTED_EVENT.event(),
+                        stack.getName(), null);
+            } else if (externalDatabase == DatabaseAvailabilityType.NONE) {
+                LOGGER.info("External database for stack {} is not requested. Start is not possible.", stack.getName());
+                result = new StartExternalDatabaseResult(stack.getId(), EXTERNAL_DATABASE_STARTED_EVENT.event(),
+                        stack.getName(), null);
+            } else if (!externalDatabaseConfig.isExternalDatabasePauseSupportedFor(CloudPlatform.valueOf(environment.getCloudPlatform()))) {
```

Comment:
should we just redbeams and if redbeams say it is not supported than we just ignore the start stop on the DB ?